pkg/importer/dataStream.go

```diff
@@ -905,13 +916,17 @@ func (d *DataStream) copy(dest string) error {
 			return err
 		}
 	} else {
-		if util.GetAvailableSpace(d.ScratchDataDir) <= int64(0) {
+		//we do not need scratch space for images with raw format.
+		// d.isIsoImage is assigned to true when it's raw format
+		if !d.isIsoImage && util.GetAvailableSpace(d.ScratchDataDir) <= int64(0) {
 			//Need scratch space but none provided.
 			return ErrRequiresScratchSpace
 		}
-		// Replace /data/target name with scratch path/target name
-		tmpDest := filepath.Join(d.ScratchDataDir, filepath.Base(dest))
-
+		tmpDest := dest
```

Comment:
n this path we have already decided we need scratch space in ALL cases, so this makes no sense. Again, simply set d.qemu=true in newDataStream and isHttpQCow2 will be true for an iso image and none of this is necessary.